services/subscriber/service.go

```diff
@@ -0,0 +1,258 @@
+			s.mu.Lock()
+			if !s.closed {
+				s.mu.Unlock()
+				break
+			}
+			s.mu.Unlock()
+			s.Update()
+		}
+	}
+
+}
+
+// start new and stop deleted subscriptions.
+func (s *Service) Update() error {
+	s.Logger.Println("updating subscriptions")
+	dbis, err := s.MetaStore.Databases()
+	if err != nil {
+		return err
+	}
+	allEntries := make(map[subEntry]bool, 0)
+	// Add in new subscriptions
+	for _, dbi := range dbis {
+		for _, rpi := range dbi.RetentionPolicies {
+			for _, si := range rpi.Subscriptions {
+				se := subEntry{
+					db:   dbi.Name,
+					rp:   rpi.Name,
+					name: si.Name,
+				}
+				allEntries[se] = true
+				if _, ok := s.subs[se]; ok {
+					continue
+				}
+				sub, err := s.createSubscription(se, si.Mode, si.Destinations)
+				if err != nil {
+					return err
+				}
+				s.subs[se] = sub
+			}
+		}
+	}
+
+	// Remove deleted subs
+	for se := range s.subs {
+		if !allEntries[se] {
+			delete(s.subs, se)
+		}
+	}
+
+	return nil
+}
+
+func (s *Service) createSubscription(se subEntry, mode string, destinations []string) (PointsWriter, error) {
+	var bm BalanceMode
+	switch mode {
+	case "ALL":
+		bm = ALL
+	case "ANY":
+		bm = ANY
+	default:
+		return nil, fmt.Errorf("unknown balance mode %q", mode)
+	}
+	writers := make([]PointsWriter, len(destinations))
+	statMaps := make([]*expvar.Map, len(writers))
+	for i, dest := range destinations {
+		u, err := url.Parse(dest)
+		if err != nil {
+			return nil, err
+		}
+		w, err := s.NewPointsWriter(*u)
+		if err != nil {
+			return nil, err
+		}
+		writers[i] = w
+		tags := map[string]string{
+			"database":         se.db,
+			"retention_policy": se.rp,
+			"name":             se.name,
+			"mode":             mode,
+			"destination":      dest,
+		}
+		key := strings.Join([]string{"subscriber", se.db, se.rp, se.name, dest}, ":")
+		statMaps[i] = influxdb.NewStatistics(key, "subscriber", tags)
+	}
+	return &balancewriter{
+		bm:       bm,
+		writers:  writers,
+		statMaps: statMaps,
+	}, nil
+}
+
+// Return channel into which write point requests can be sent.
+func (s *Service) Points() chan<- *cluster.WritePointsRequest {
+	return s.points
+}
+
+// read points off chan and write them
+func (s *Service) writePoints() {
+	defer s.wg.Done()
+	for p := range s.points {
+		for se, sub := range s.subs {
+			if p.Database == se.db && p.RetentionPolicy == se.rp {
+				err := sub.WritePoints(p)
+				if err != nil {
+					s.Logger.Println(err)
+					s.statMap.Add(statWriteFailures, 1)
+				}
+			}
+		}
+		s.statMap.Add(statPointsWritten, int64(len(p.Points)))
+	}
+}
+
+type BalanceMode int
+
+const (
+	ALL BalanceMode = iota
+	ANY
+)
+
+// balances writes across PointsWriters according to BalanceMode
+type balancewriter struct {
+	bm       BalanceMode
+	writers  []PointsWriter
+	statMaps []*expvar.Map
+	i        int
+}
+
+func (b *balancewriter) WritePoints(p *cluster.WritePointsRequest) error {
+	var lastErr error
+	for range b.writers {
+		i := b.i
+		w := b.writers[i]
```

Comment:
The module code rotates through destinations, otherwise each write would always try the first destination first and it would get a disproportionate amount of data.